### ISSUE TITLE
ci-operator: cp: preserving times for '...' Operation not permitted

### ISSUE THREAD

**Issue by tnozicka** [openshift/release#1584]
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-openshift-apiserver-operator/3/pull-ci-openshift-cluster-openshift-apiserver-operator-master-verify/16

This is likely related to the CI cluster and possibly OpenShift security restriction config or broken volumes. I run the ci-operator locally with that config on clusterup and it passed.

We have went around it by avoiding the code path with `cp -a` but it might worth looking into.

/kind bug
/cc @stevekuznetsov @smarterclayton

**Comment by smarterclayton** [comment]
No other job seems to have this.  Why is your job special?

**Comment by tnozicka** [comment]
No idea, maybe they don't use `cp -a` in `/tmp/`

**Comment by tnozicka** [comment]
We are still seeing this on unhappy paths. This time in `openshift/api` which is not related to our new operator repos. We can probably add the hack there as well but this shows that something might be wrong with the CI cluster.

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_api/109/pull-ci-openshift-api-master-verify/10

/cc @mfojtik 
/reopen

**Comment by tnozicka** [comment]
Might be an issue with how CI operator setups the repo in the pod. For reference, this is the commit removing the `cp -a` lines to go around it https://github.com/openshift/cluster-openshift-apiserver-operator/pull/6/commits/08ce259dc467c3719ba7c4e6da2375be08185250#diff-dc2915041d23416c6ec10531445397baL35

I've tried simple `cp -a` in api.ci and it worked fine

**Comment by aveshagarwal** [comment]
I am running into this issue as well (https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-kube-scheduler-operator/35/pull-ci-openshift-cluster-kube-scheduler-operator-master-verify/3). 

The issue, as I understand, is that containers are being run with some random non-root user id, and that user belongs to the root group. All files in the container are owned by `root`, and `cp -a` tries to preserve all attributes (including time stamps, mode, owner etc) and when it overwrites files belonging to root user by files with (random) non-root user id), it fails. It can be reproduced on locally by running source image as follows:

```
$ sudo docker run -u 30000 -it registry.svc.ci.openshift.org/ci-op-4gj6hkr9/pipeline:src
bash-4.2$ touch test1
bash-4.2$ ls -al
-rw-rwxr--. 1 root  root 11357 Nov 28 19:46 LICENSE
-rw-r--r--. 1 30000 root     0 Nov 29 01:40 test1
bash-4.2$ cp -a test1 LICENSE
cp: preserving times for 'LICENSE': Operation not permitted
```

**The thing is that simple `cp` works** Also `cp --preserve=ownership` works inside container too. But `cp --preserve=timestamps` or `cp --preserve=mode` fails.

Infact somewhat similar behavior can be seen on host system too, lets say create two files test1 and test2 on a system where test1 is owned by root and test2 is owner by some user (avagarwa in this case) and both files belong to root group.

```
-rw-r--r--.  1 root     root           0 Nov 28 20:47 test1
-rw-r--r--.  1 avagarwa root           0 Nov 28 20:47 test2
$ cp -a test2 test1
cp: cannot create regular file 'test1': Permission denied
$ cp --preserve=timestamps test2 test1
cp: cannot create regular file 'test1': Permission denied
$ cp --preserve=mode test2 test1
cp: cannot create regular file 'test1': Permission denied
$ cp --preserve=ownership test2 test1
cp: cannot create regular file 'test1': Permission denied
```
**The strange thing is that `cp --preserve=ownership` works inside container but not on host system.**

Though `cp -a` fails here too, but the interesting thing is that the error is different. 

Anyway, in summary, the behavior inside container or a host system is somewhat similar except the different error messages.

@smarterclayton @sjenning

**Comment by tnozicka** [comment]
@mfojtik seen it here today https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_api/267/pull-ci-openshift-api-master-verify/548

**Comment by tnozicka** [comment]
(That job was using `cp -a` as well https://github.com/openshift/api/blob/80eef12477e5367bfe16a17731bc5fbf80a09e74/hack/verify-protobuf.sh#L17)

**Comment by LorbusChris** [comment]
Hitting this here: https://github.com/openshift/api/pull/301
